lib/plugins/ar2.js

```diff
@@ -12,133 +23,75 @@ function init() {
   ar2.pluginType = 'forecast';
 
-  var BG_REF = 140; //Central tendency
-  var BG_MIN = 36; //Not 39, but why?
-  var BG_MAX = 400;
-  var WARN_THRESHOLD = 0.05;
-  var URGENT_THRESHOLD = 0.10;
-
-  var ONE_HOUR = 3600000;
-  var ONE_MINUTE = 60000;
-  var FIVE_MINUTES = 300000;
-  var TEN_MINUTES =  600000;
-
+  ar2.setProperties = function setProperties (sbx) {
+    sbx.offerProperty('ar2', function setAR2 ( ) {
+      return {
+        forecast: ar2.forecast(sbx.data.sgvs)
+        , rawForecast: rawForecast(sbx.data.sgvs, sbx)
+      };
+    });
+  };
 
   ar2.checkNotifications = function checkNotifications(sbx) {
 
     var lastSGVEntry = _.last(sbx.data.sgvs);
 
-    var result = {};
-
-    if (lastSGVEntry && Date.now() - lastSGVEntry.x < TEN_MINUTES) {
-      result = ar2.forcastAndCheck(sbx.data.sgvs);
-    }
-
-    var usingRaw = false;
-    if (!result.trigger && sbx.extendedSettings.useRaw) {
-      var cal = _.last(sbx.data.cals);
-      if (cal) {
-        var rawSGVs = _.map(_.takeRight(sbx.data.sgvs, 2), function eachSGV (sgv) {
-          var rawResult = rawbg.calc(sgv, cal);
-
-          //ignore when raw is 0, and use BG_MIN if raw is lower
-          var rawY = rawResult === 0 ? 0 : Math.max(rawResult, BG_MIN);
-
-          return { x: sgv.x, y: rawY };
-        });
-        result = ar2.forcastAndCheck(rawSGVs, true);
-        usingRaw = true;
-      }
+    if (!lastSGVEntry || Date.now() - lastSGVEntry.x > TEN_MINUTES) {
+      return;
     }
 
-    if (result.trigger) {
-
-      var predicted = _.map(result.forecast.predicted, function(p) { return sbx.scaleBg(p.y) } );
-
-      var first = _.first(predicted);
-      var last = _.last(predicted);
-      var avg = _.sum(predicted) / predicted.length;
+    var prop = sbx.properties.ar2;
 
-      var max = _.max([first, last, avg]);
-      var min = _.min([first, last, avg]);
+    var result = checkForecast(prop.forecast) || checkForecast(prop.rawForecast, true);
 
-      var rangeLabel = '';
-      var eventName = '';
-      if (max > sbx.scaleBg(sbx.thresholds.bg_target_top)) {
-        rangeLabel = 'HIGH';
-        eventName = 'high';
-        if (!result.pushoverSound) { result.pushoverSound = 'climb'; }
-      } else if (min < sbx.scaleBg(sbx.thresholds.bg_target_bottom)) {
-        rangeLabel =  'LOW';
-        eventName = 'low';
-        if (!result.pushoverSound) { result.pushoverSound = 'falling'; }
-      } else {
-        rangeLabel = 'Check BG';
-      }
+    if (result) {
+      var eventName = selectEventType(result, sbx);
+      var rangeLabel = eventName ? eventName.toUpperCase() : 'Check BG';
 
-      var title = sbx.notifications.levels.toString(result.level) + ', ' + rangeLabel;
-      if (lastSGVEntry.y > sbx.thresholds.bg_target_bottom && lastSGVEntry.y < sbx.thresholds.bg_target_top) {
-        title += ' predicted';
+      if (eventName && !result.pushoverSound) {
+        result.pushoverSound = eventName == 'high' ? 'climb' : 'falling';
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768517600/issues/source?bid=2452379&fileBranchId=2457555#l52)